src/main/java/org/beanpod/switchboard/controller/DecoderController.java

```diff
@@ -81,4 +80,15 @@ public ResponseEntity createDecoder(@RequestBody @Valid DecoderEntity decoderEnt
         }
         return ResponseEntity.ok("Decoder with serial number " + serialNumber + " Deleted");
     }
+
+    @PutMapping
+    public ResponseEntity<DecoderDTO> updateDecoder( @RequestBody DecoderDTO decoderDTO){
```

Comment:
Some strange behaviour i came across when manually testing this endpoint
- It's possible to update details about a device through updating its decoder. Do we want to keep that behaviour?
- It's possible to change the serial number of a device through this endpoint, without changing the serial number of the decoder (and without changing the serial number of any encoder that might be associated with the device, too). We probably want to update all the serial numbers to match, OR prevent changing serial numbers altogether.

Reply:
good catch, ill look into the fix

Reply:
We decided to create a bug for this once the PR is merged